src/Raven.Server/Documents/Handlers/Processors/AbstractIndexHandlerProcessorForGetDatabaseIndexStatistics.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+using Raven.Client.Http;
+using Raven.Server.Json;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors
+{
+    internal abstract class AbstractIndexHandlerProcessorForGetDatabaseIndexStatistics<TRequestHandler, TOperationContext> : AbstractHandlerReadProcessor<IndexStats[], TRequestHandler, TOperationContext>
```

Comment:
Suggestion regarding the names:

- @ppekrol maybe instead of `AbstractHandlerReadProcessor` - `AbstractReadRequestProcessor`?
- @shiranshalom if we changed the above then maybe this class could be `AbstractIndexStatisticsRequestProcessor` and its 2 implementations would be:

   - `DatabaseIndexStatisticsRequestProcessor`
   - `ShardedIndexStatisticsRequestProcessor`
   
   What do you think?

Reply:
Actually maybe using `IndexStats` instead of `IndexStatistics` would be more consistent with the endpoint paths- `/indexes/stats`

Reply:
I like those naming conventions much better!

Reply:
but maybe `IndexesStats` to match the path

Reply:
Name pattern is Abstract<HandlerName>ProcessorFor<MethodNameInHandler>